File: src/main/java/com/fhkdemo/multitenant/SpringKeycloakMultiTenantApplication.java
package com.fhkdemo.multitenant;

import com.fhkdemo.multitenant.configs.PathBasedConfigResolver;
import org.keycloak.adapters.KeycloakConfigResolver;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
import org.springframework.context.annotation.Bean;

@SpringBootApplication
public class SpringKeycloakMultiTenantApplication {
    public static void main(String[] args) {
        SpringApplication.run(SpringKeycloakMultiTenantApplication.class, args);
    }

    /**
     * Required to handle spring boot configurations
     * 
     * @return
     */
    @Bean
    @ConditionalOnMissingBean(PathBasedConfigResolver.class)
    public KeycloakConfigResolver keycloakConfigResolver() {
        return new PathBasedConfigResolver();
    }
}

File: src/main/kotlin/com/fhkdemo/multitenant/SpringKeycloakMultiTenantApplication.kt
package com.fhkdemo.multitenant

import org.keycloak.adapters.KeycloakConfigResolver
import org.keycloak.adapters.springboot.KeycloakSpringBootConfigResolver
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Import

@SpringBootApplication(exclude = [DataSourceAutoConfiguration::class])
//@Import(KeycloakSpringBootConfigResolver::class)
class SpringKeycloakMultiTenantApplication {
	@Bean
	@ConditionalOnMissingBean(PathBasedConfigResolver::class)
	fun keycloakConfigResolver(): KeycloakConfigResolver {
		return PathBasedConfigResolver()
	}
}

fun main(args: Array<String>) {
	runApplication<SpringKeycloakMultiTenantApplication>(*args)
}
//{
//
//	@Bean
//	@ConditionalOnMissingBean(PathBasedConfigResolver::class)
//	fun keycloakConfigResolver(): KeycloakConfigResolver {
//		return PathBasedConfigResolver()
//	}
//
//	companion object {
//		@JvmStatic
//		fun main(args: Array<String>) {
//			SpringApplication.run(SmsgatewaymgntApplication::class.java, *args)
//		}
//	}
//}
